language/move-lang/tests/move_check/typing/assign_unpack_references.move

```diff
@@ -0,0 +1,28 @@
+module M {
+    struct S { f: u64 }
+    struct R { s1: S, s2: S }
+
+    t0() {
+        let f;
+        let s2;
+        R { s1: S { f }, s2 } = R { s1: S{f: 0}, s2: S{f: 1} };
+        f = 0;
+        s2 = S { f: 0 }
```

Comment:
Do you have a similar test to this but where you annotate with the wrong types in the declaration?

Reply:
... maybe I'll double check